Day 3. Homerowk

eclimov opened the pull request:
- Entities' relations (+ "add", "remove", "get" methods)
- Lifecycle callbacks
- Migrations
- Fixtures (+ dependencies)
- Faker

eclimov committed:
Entities' relations created
- Migrations generated and executed

eclimov committed:
Lifecycle callbacks added
- Instead of adding the callback to "createdAt" columns, changed the entities on DB schema level, because it is considered a better practice

eclimov committed:
Fixtures created
- Using Faker
- Using dependencies
- "created_at" column default value logic changed: now defining it at Entity level. Reason: DB unexpected behavior, when adding fixtures
- Migrations added

gregurco commented on src/DataFixtures/AffiliateFixtures.php:
```diff
@@ -0,0 +1,41 @@
+            $affiliate = new Affiliate();
+            $affiliate->setUrl($faker->url);
+            $affiliate->setEmail($faker->email);
+            $affiliate->setToken($faker->password(6,6));
```
code style fix, should be:
`$faker->password(6, 6)`
(space after **,**)

gregurco commented on src/DataFixtures/AffiliateFixtures.php:
```diff
@@ -0,0 +1,41 @@
+            $affiliate->setEmail($faker->email);
+            $affiliate->setToken($faker->password(6,6));
+            $affiliate->setActive($faker->boolean);
+            $affiliate->addCategory($this->getReference('category_'.$i));
```
also code style fix (space before and after **.**):
`$this->getReference('category_' . $i)`

gregurco commented on src/DataFixtures/AffiliateFixtures.php:
```diff
@@ -0,0 +1,41 @@
+     */
+    public function getDependencies(): array
+    {
+        return array(
```
use `[]` instead of `array()`

gregurco commented on src/DataFixtures/CategoryFixtures.php:
```diff
@@ -0,0 +1,27 @@
+    public function load(ObjectManager $manager): void
+    {
+        $faker = Faker\Factory::create('en_US');
+        for ($i = 0; $i < 10; $i++) {
```
add new empty line before `for`. It will do code more readable

gregurco commented on src/DataFixtures/CategoryFixtures.php:
```diff
@@ -0,0 +1,27 @@
+            $category = new Category();
+            $category->setName(ucfirst($faker->word));
+            $manager->persist($category);
+            $this->addReference('category_'.$i, $category);
```
also code style

gregurco commented on src/Entity/Affiliate.php:
```diff
@@ -49,6 +51,47 @@ class Affiliate
+     */
+    public function addCategory(Category $category): Affiliate
+    {
+        $this->categories->add($category);
```
better to check if element if not already added:
```
if (!$this->categories->contains($category)) {
    $this->categories->add($category);
}
```

eclimov committed:
Code style updated, according to best practices. Logic slightly improved
- Space added between parameters' lists
- "array(...)" replaced with "[...]"
- Empty line added before "for()" loops
- Spaces added before and after the dot in string concatenation
- Added check for existence in "add" methods of entities

eclimov commented:
Fixed

gregurco commented on src/Entity/Affiliate.php:
```diff
@@ -49,6 +51,51 @@ class Affiliate
+     */
+    public function addCategory(Category $category): Affiliate
+    {
+        if(!$this->categories->contains($category))
```
code style:
```
if (!$this->categories->contains($category)) {
    $this->categories->add($category);
}
```

gregurco commented on src/Entity/Category.php:
```diff
@@ -23,6 +25,57 @@ class Category{
+     */
+    public function addJob(Job $job): Category
+    {
+        if(!$this->jobs->contains($job))
```
also code style

eclimov committed:
Code style problem fixed
- "if" instruction's brace position

eclimov commented:
Fixed
